lib/tapioca/compilers/dsl/active_record_identity_cache.rb

```diff
@@ -108,7 +108,7 @@ def decorate(root, constant)
         sig { override.returns(T::Enumerable[Module]) }
         def gather_constants
           ::ActiveRecord::Base.descendants.select do |klass|
-            klass < IdentityCache
+            klass < IdentityCache || klass < IdentityCache::WithoutPrimaryIndex
```

Comment:
I think the first always includes the second, so it should be enough to just check for the second.

Reply:
Oh yeah it is enough to only check for the second. The `<` is not ideal to me then. I thought it would effectively be "recursive" when checking for a subclass.

Reply:
`<` does not mean subclass, though. It looks to see if the given module is in the list of ancestors. There is nothing recursive that is needed, every class that pulls in `IdentityCache` also pulls in `IdentityCache::WithoutPrimaryIndex` into its list of ancestors, as well.

Reply:
Yes, I can't figure out what I was thinking when I wrote that comment now 😄